ts/components/leftpane/overlay/choose-action/ContactsListWithBreaks.tsx

```diff
@@ -0,0 +1,128 @@
+    display: none;
+  }
+
+  .module-conversation-list-item __buttons {
+    display: flex;
+
+    .session-button {
+      font-size: var(--font-size-xs);
+      padding: 6px;
+      height: auto;
+      margin: 0px;
+      line-height: 14px;
+    }
+  }
+`;
+
+const renderRow = (props: ListRowProps) => {
+  const { index, key, style, parent } = props;
+
+  // ugly, but it seems react-virtualized does not support very well functional components just yet
+  // https://stackoverflow.com/questions/54488954/how-to-pass-prop-into-rowrender-of-react-virtualized
+  const directContactsByNameWithBreaks = (parent as any).props
+    .directContactsByNameWithBreaks as Array<DirectContactsByNameType | string>;
+  const item = directContactsByNameWithBreaks?.[index];
+  if (!item) {
+    return null;
+  }
+
+  if (isString(item)) {
+    return <ContactRowBreak style={style} key={key} char={item} />;
+  }
+
+  return <ContactRow style={style} key={key} {...item} />;
+};
+
+const unknownSection = 'unknown';
+
+const ContactListItemSection = () => {
+  const directContactsByName = useSelector(getDirectContactsByName);
+
+  if (!directContactsByName) {
+    return null;
+  }
+
+  // add a break wherever needed
+  let currentChar = '';
+  // if the item is a string we consider it to be a break of that string
+  const directContactsByNameWithBreaks: Array<DirectContactsByNameType | string> = [];
+  directContactsByName.forEach(m => {
+    if (m.displayName && m.displayName[0] !== currentChar) {
+      currentChar = m.displayName[0];
+      directContactsByNameWithBreaks.push(currentChar.toUpperCase());
+    } else if (!m.displayName && currentChar !== unknownSection) {
+      currentChar = unknownSection;
+      directContactsByNameWithBreaks.push(window.i18n('unknown'));
+    }
+    directContactsByNameWithBreaks.push(m);
+  });
+
+  const length = Number(directContactsByNameWithBreaks.length);
+
+  return (
+    <StyledLeftPaneList key={0} style={{ width: '100%' }}>
+      <AutoSizer>
+        {({ height }) => {
+          return (
+            <List
+              className="module-left-pane__virtual-list"
+              height={height}
+              rowCount={length}
+              rowHeight={
+                (params: Index) =>
+                  isString(directContactsByNameWithBreaks[params.index]) ? 30 : 64 // should also be changed in `ContactRowBreak`
+              }
+              directContactsByNameWithBreaks={directContactsByNameWithBreaks}
+              rowRenderer={renderRow}
+              width={300} // the same as session-left-pane-width
+              autoHeight={false}
+            />
+          );
+        }}
+      </AutoSizer>
+    </StyledLeftPaneList>
+  );
+};
+
+const ContactsTitle = () => {
+  const contactsCount = useSelector(getDirectContactsCount);
+  if (contactsCount <= 0) {
+    return null;
+  }
+
+  return (
+    <StyledChooseActionTitle tabIndex={0}>{window.i18n('contactsHeader')}</StyledChooseActionTitle>
```

Comment:
Again, weird that the indentation isn't formatted here

Reply:
I can't force it. We would need to add eslint at some point and update prettier rules, but it's not going to happen on this PR :)

Reply:
👍